pages/robot/tabs/components/index.php

```diff
@@ -284,6 +563,254 @@ function render_components(data) {
                         status_icon(component.calibration.completed, true, null)
                     ) : ''
             ) : '';
+            let verification_test_button = "";
+            // if (component.detected) {
+            if (component.supported && component.test_service_name !== "") {
+                let id_str_name = name.replaceAll(' ', '-');
+                verification_test_button = '<button type="button" disabled="true" id="modal-btn-' + id_str_name + '" class="btn btn-info" data-toggle="modal" data-target="' + '#modal-' + id_str_name + '">Test Hardware</button>';
```

Comment:
BTW, you can use the function `String.format()`, for example, this would become:
```
verification_test_button = '<button type="button" disabled="true" id="modal-btn-{0}" class="btn btn-info" data-toggle="modal" data-target="#modal-{1}">Test Hardware</button>'.format(id_str_name, id_str_name);
```